site/_posts/2021-09-16-how-to-authenticate-your-react-app.md

```diff
@@ -0,0 +1,1087 @@
+
+* You need to have either a FusionAuth Cloud account or install their self-hosted version, available for free. Follow the [FusionAuth 5-Minute Setup Guide](/docs/v1/tech/5-minute-setup-guide/) to get started with your installation.  
+* Your Node version should be at least 14.17.6. We use this version in this tutorial.  
+* We use React 17.0.2 and suggest you use the same or most recent stable build.
+
+Once you have installed all the required components, log into your FusionAuth instance by visiting `localhost:9011`.
+
+We will do the following in this tutorial:
+
+- [Set up FusionAuth.](#configuring-fusionAuth)
+- [Create a basic React UI.](#create-the-react-ui)
+- [Create an Express backend server.](#create-the-express-server)
+- [Log a user in.](#logging-in)
+- [Display A User's Data.](#display-a-users-data)
+- [Log a user out.](#logging-out)
+
+All of this is going to happen using the OAuth 2.0 Authorization Code grant.  
+
+## Configuring FusionAuth
+
+You have two options to configure FusionAuth, either manually or with Kickstart.
+
+### Use Kickstart
+
+Instead of manually setting up FusionAuth using the admin UI, you can use Kickstart. This tool allows you to get going quickly if you have a fresh installation of FusionAuth. Learn more about how to use [Kickstart](/docs/v1/tech/installation-guide/kickstart/). Here's an example [Kickstart file](https://github.com/FusionAuth/fusionauth-example-kickstart/blob/master/example-apps/path-to-kickstart-here) which sets up FusionAuth for this tutorial.
+
+You'll need to edit that Kickstart file and update the application name, API Key, and redirect URLs with your specific app's data.
+
+### Manual configuration
+
+If you choose not to use kickstart, you must configure an application in FusionAuth as well as link the user account with which you’ll log in.
+
+*"But wait a minute!" you may ask. "Why do I have to make another app in FusionAuth? Don't I already have a React App?  I thought I could just import FusionAuth functionality into my React app?"*
+
+You can't directly import FusionAuth functionality into your JavaScript or React projects.  Instead,  FusionAuth is a separate server that developers can interact with via API calls.  So, instead of being something like a JavaScript library that you include into your own JavaScript projects and pull functionality from, FusionAuth actually is an entire application that you configure to work with your React app and server of your choosing via API calls. (Those API calls can be made by a library such as `passport.js`, however.)  Think of it like a MongoDB or Postgres database, but which handles auth, rather than storing data.   This does have some benefits, however, such as far more layers of security, and more control over using FusionAuth as a user data store.  You can learn more from our [expert advice](/learn/expert-advice/identity-basics/complete-authentication-system/).
```

Comment:
```suggestion
You can't directly import FusionAuth functionality into your JavaScript or React projects. Instead, FusionAuth is a separate server that developers can interact with via API calls. So, instead of being something like a JavaScript library that you include into your own JavaScript projects and pull functionality from, FusionAuth actually is an entire application that you configure to work with your React app and server of your choosing via API calls. (Those API calls can be made by a library such as `passport.js`, however.) Think of it like a MongoDB or Postgres database, but which handles auth, rather than storing data. This does have some benefits, however, such as far more layers of security, and more control over using FusionAuth as a user data store. You can learn more from our [expert advice](/learn/expert-advice/identity-basics/complete-authentication-system/).
```